unit-test/background/tracker-utils-tests.js

```diff
@@ -0,0 +1,275 @@
+        },
+        {
+            name: 'referrer & target are safelisted',
+            referrer: 'http://siteA.com',
+            target: 'http://siteB.com',
+            safelist: {'sitea.com': true, 'siteb.com': true},
+            expectedReferrer: undefined
+        },
+        {
+            name: 'subdomain of safelisted target',
+            referrer: 'http://siteA.com',
+            target: 'http://subdomain.siteB.com',
+            safelist: {'siteb.com': true},
+            expectedReferrer: undefined
+        }
+    ]
+    it('Should not modify referrer when either site is safe listed by the user', () => {
+        for (let test of referrerUserSafelistTests) {
+            settingsObserver.and.returnValue(test.safelist)
+            expect(trackerutils.truncateReferrer(test.referrer, test.target)).withContext(`safelist: ${test.name}`).toEqual(test.expectedReferrer)
+        }
+    })
+
+    const referrerSafelistTests = [
+        {
+            name: 'referrer is safelisted',
+            referrer: 'http://test.com',
+            target: 'http://siteA.com',
+            expectedReferrer: undefined
+        },
+        {
+            name: 'target is safelisted',
+            referrer: 'http://siteA.com',
+            target: 'http://test.com',
+            expectedReferrer: undefined
+        },
+        {
+            name: 'safe listed entry has a path',
+            referrer: 'http://testing.com/some/path/here',
+            target: 'http://sitea.com',
+            expectedReferrer: undefined
+        },
+        {
+            name: 'subdomain of safelisted target',
+            referrer: 'http://siteA.com',
+            target: 'http://subdomain.testing.com/some/path?option=yes&option2=no',
+            expectedReferrer: undefined
+        }
+    ]
+    it('Should not modify referrer when either site is safe listed by the global referrer safe list', () => {
+        for (let test of referrerSafelistTests) {
+            settingsObserver.and.returnValue(undefined)
+            expect(trackerutils.truncateReferrer(test.referrer, test.target)).withContext(`test: ${test.name}`).toEqual(test.expectedReferrer)
+        }
+    })
+
+    const referrerTruncationTests = [
+        {
+            name: 'Simple truncation test',
+            referrer: 'http://siteA.com/article/1',
+            target: 'http://siteB.com',
+            expectedReferrer: 'http://sitea.com'
+        },
+        {
+            name: 'target is a tracker, and referrer has a subdomain',
+            referrer: 'http://subdomain.siteA.com/article/1',
+            target: 'https://google-analytics.com/some/path',
+            expectedReferrer: 'http://sitea.com'
+        },
+        {
+            name: 'target is not a tracker, referrer should keep subdomain',
+            referrer: 'http://subdomain.siteA.com/article/1',
+            target: 'http://siteB.comm',
+            expectedReferrer: 'http://subdomain.sitea.com'
+        }
```

Comment:
lets also test stuff with multiple subdomain "levels" (a.b.c.example.com) and that "www." thing